### ISSUE TITLE
Calendar doesn't show Sonarr/Radarr elements

### ISSUE THREAD

**Issue by Bencici** [causefx/Organizr#1943]
###### Operating System: Windows Server 2022
<hr>

##### Problem Description:
Calendar won't show Sonarr/Radarr episodes/films
Started the server after it was off for more than a year, updated everything and I can't make it work. Did the calendar item changed the way it worked ?
I also tried using the iCal links from Sonarr/Radarr but it stays desperately empty.

I went from the master branch to  the develop one, but the issue persists.
All Organizr files are present, and there don't seem to be any permission issues.

<hr>

##### Reproduction Steps:
Open Organizrr main page
or
Query "http://adress/api/v2/homepage/calendar" with Powershell

I pull events from 30 days before to 60 days after today and yes I can see them on Sonarr/Radarr calendars.

<hr>

#### Errors on screen?  If so paste here:

```
Error 500 on "http://ADRESS/api/v2/homepage/calendar" in the browser console :

Type: ValueError
Code: 0
Message: Path cannot be empty
File: C:\nginx\www\organizr\html\api\functions\organizr-functions.php
Line: 568
Trace

#0 C:\nginx\www\organizr\html\api\functions\organizr-functions.php(568): copy('', 'C:\\nginx\\www\\or...', Resource id #476)
#1 C:\nginx\www\organizr\html\api\homepage\sonarr.php(293): Organizr->cacheImage(NULL, 391020)
#2 C:\nginx\www\organizr\html\api\homepage\sonarr.php(224): Organizr->formatSonarrCalendar(Array, 0)
#3 C:\nginx\www\organizr\html\api\homepage\calendar.php(95): Organizr->getSonarrCalendar('2023-06-10', '2023-09-08')
#4 C:\nginx\www\organizr\html\api\v2\routes\homepage.php(13): Organizr->getCalendar()
#5 C:\nginx\www\organizr\html\api\vendor\slim\slim\Slim\Handlers\Strategies\RequestResponse.php(43): {closure}(Object(Slim\Psr7\Request), Object(Slim\Psr7\Response), Array)
#6 C:\nginx\www\organizr\html\api\vendor\slim\slim\Slim\Routing\Route.php(384): Slim\Handlers\Strategies\RequestResponse->__invoke(Object(Closure), Object(Slim\Psr7\Request), Object(Slim\Psr7\Response), Array)
#7 C:\nginx\www\organizr\html\api\vendor\slim\slim\Slim\MiddlewareDispatcher.php(81): Slim\Routing\Route->handle(Object(Slim\Psr7\Request))
#8 C:\nginx\www\organizr\html\api\vendor\slim\slim\Slim\MiddlewareDispatcher.php(81): Slim\MiddlewareDispatcher->handle(Object(Slim\Psr7\Request))
#9 C:\nginx\www\organizr\html\api\vendor\slim\slim\Slim\Routing\Route.php(341): Slim\MiddlewareDispatcher->handle(Object(Slim\Psr7\Request))
#10 C:\nginx\www\organizr\html\api\vendor\slim\slim\Slim\Routing\RouteRunner.php(84): Slim\Routing\Route->run(Object(Slim\Psr7\Request))
#11 C:\nginx\www\organizr\html\api\vendor\slim\slim\Slim\Middleware\RoutingMiddleware.php(59): Slim\Routing\RouteRunner->handle(Object(Slim\Psr7\Request))
#12 C:\nginx\www\organizr\html\api\vendor\slim\slim\Slim\MiddlewareDispatcher.php(147): Slim\Middleware\RoutingMiddleware->process(Object(Slim\Psr7\Request), Object(Slim\Routing\RouteRunner))
#13 C:\nginx\www\organizr\html\api\vendor\slim\slim\Slim\Middleware\ErrorMiddleware.php(107): Psr\Http\Server\RequestHandlerInterface@anonymous->handle(Object(Slim\Psr7\Request))
#14 C:\nginx\www\organizr\html\api\vendor\slim\slim\Slim\MiddlewareDispatcher.php(147): Slim\Middleware\ErrorMiddleware->process(Object(Slim\Psr7\Request), Object(Psr\Http\Server\RequestHandlerInterface@anonymous))
#15 C:\nginx\www\organizr\html\api\v2\index.php(92): Psr\Http\Server\RequestHandlerInterface@anonymous->handle(Object(Slim\Psr7\Request))
#16 C:\nginx\www\organizr\html\api\vendor\slim\slim\Slim\MiddlewareDispatcher.php(313): {closure}(Object(Slim\Psr7\Request), Object(Psr\Http\Server\RequestHandlerInterface@anonymous))
#17 C:\nginx\www\organizr\html\api\vendor\slim\slim\Slim\MiddlewareDispatcher.php(81): Psr\Http\Server\RequestHandlerInterface@anonymous->handle(Object(Slim\Psr7\Request))
#18 C:\nginx\www\organizr\html\api\vendor\slim\slim\Slim\App.php(215): Slim\MiddlewareDispatcher->handle(Object(Slim\Psr7\Request))
#19 C:\nginx\www\organizr\html\api\vendor\slim\slim\Slim\App.php(199): Slim\App->handle(Object(Slim\Psr7\Request))
#20 C:\nginx\www\organizr\html\api\v2\index.php(151): Slim\App->run()
#21 {main}
```
![image](https://github.com/causefx/Organizr/assets/4329610/3492583f-a00a-49b9-ba37-7bb8c213e3f7)

If I type http://ADRESS/api/v2/homepage/calendar in my browser :
![BdxysQg](https://github.com/causefx/Organizr/assets/4329610/7003df55-1320-47f9-9796-6efeaa808cab)

**Comment by causefx** [comment]
can you try going to these 2 URLs in your browser:

api/v2/homepage/sonarr/calendar
api/v2/homepage/radarr/calendar

**Comment by Bencici** [comment]
> can you try going to these 2 URLs in your browser:
> 
> api/v2/homepage/sonarr/calendar api/v2/homepage/radarr/calendar

Radarr show a JSON output, so it looks like it works. For Sonarr I have the same error :
```
Type: ValueError
Code: 0
Message: Path cannot be empty
File: C:\nginx\www\organizr\html\api\functions\organizr-functions.php
Line: 568
Trace

#0 C:\nginx\www\organizr\html\api\functions\organizr-functions.php(568): copy('', 'C:\\nginx\\www\\or...', Resource id #476)
#1 C:\nginx\www\organizr\html\api\homepage\sonarr.php(293): Organizr->cacheImage(NULL, 391020)
#2 C:\nginx\www\organizr\html\api\homepage\sonarr.php(224): Organizr->formatSonarrCalendar(Array, 0)
#3 C:\nginx\www\organizr\html\api\v2\routes\homepage.php(141): Organizr->getSonarrCalendar()
#4 C:\nginx\www\organizr\html\api\vendor\slim\slim\Slim\Handlers\Strategies\RequestResponse.php(43): {closure}(Object(Slim\Psr7\Request), Object(Slim\Psr7\Response), Array)
#5 C:\nginx\www\organizr\html\api\vendor\slim\slim\Slim\Routing\Route.php(384): Slim\Handlers\Strategies\RequestResponse->__invoke(Object(Closure), Object(Slim\Psr7\Request), Object(Slim\Psr7\Response), Array)
#6 C:\nginx\www\organizr\html\api\vendor\slim\slim\Slim\MiddlewareDispatcher.php(81): Slim\Routing\Route->handle(Object(Slim\Psr7\Request))
#7 C:\nginx\www\organizr\html\api\vendor\slim\slim\Slim\MiddlewareDispatcher.php(81): Slim\MiddlewareDispatcher->handle(Object(Slim\Psr7\Request))
#8 C:\nginx\www\organizr\html\api\vendor\slim\slim\Slim\Routing\Route.php(341): Slim\MiddlewareDispatcher->handle(Object(Slim\Psr7\Request))
#9 C:\nginx\www\organizr\html\api\vendor\slim\slim\Slim\Routing\RouteRunner.php(84): Slim\Routing\Route->run(Object(Slim\Psr7\Request))
#10 C:\nginx\www\organizr\html\api\vendor\slim\slim\Slim\Middleware\RoutingMiddleware.php(59): Slim\Routing\RouteRunner->handle(Object(Slim\Psr7\Request))
#11 C:\nginx\www\organizr\html\api\vendor\slim\slim\Slim\MiddlewareDispatcher.php(147): Slim\Middleware\RoutingMiddleware->process(Object(Slim\Psr7\Request), Object(Slim\Routing\RouteRunner))
#12 C:\nginx\www\organizr\html\api\vendor\slim\slim\Slim\Middleware\ErrorMiddleware.php(107): Psr\Http\Server\RequestHandlerInterface@anonymous->handle(Object(Slim\Psr7\Request))
#13 C:\nginx\www\organizr\html\api\vendor\slim\slim\Slim\MiddlewareDispatcher.php(147): Slim\Middleware\ErrorMiddleware->process(Object(Slim\Psr7\Request), Object(Psr\Http\Server\RequestHandlerInterface@anonymous))
#14 C:\nginx\www\organizr\html\api\v2\index.php(92): Psr\Http\Server\RequestHandlerInterface@anonymous->handle(Object(Slim\Psr7\Request))
#15 C:\nginx\www\organizr\html\api\vendor\slim\slim\Slim\MiddlewareDispatcher.php(313): {closure}(Object(Slim\Psr7\Request), Object(Psr\Http\Server\RequestHandlerInterface@anonymous))
#16 C:\nginx\www\organizr\html\api\vendor\slim\slim\Slim\MiddlewareDispatcher.php(81): Psr\Http\Server\RequestHandlerInterface@anonymous->handle(Object(Slim\Psr7\Request))
#17 C:\nginx\www\organizr\html\api\vendor\slim\slim\Slim\App.php(215): Slim\MiddlewareDispatcher->handle(Object(Slim\Psr7\Request))
#18 C:\nginx\www\organizr\html\api\vendor\slim\slim\Slim\App.php(199): Slim\App->handle(Object(Slim\Psr7\Request))
#19 C:\nginx\www\organizr\html\api\v2\index.php(151): Slim\App->run()
#20 {main}
```

**Comment by causefx** [comment]
It seems Sonarr doesn't have or has a wrong url for an image...

What version of Sonarr?

```php
			foreach ($child['series']['images'] as $image) {
				if ($image['coverType'] == "fanart") {
					$fanart = $image['url'];
				}
			}
```

Can you get me the output from your sonarr calendar url?
`http://localhost:8989/api/v3/calendar`

**Comment by Bencici** [comment]
> It seems Sonarr doesn't have or has a wrong url for an image...
> 
> What version of Sonarr?

Sonarr v4.0.0.552

> Can you get me the output from your sonarr calendar url? `http://localhost:8989/api/v3/calendar`

```
[
  {
    "seriesId": 57,
    "tvdbId": 9686416,
    "episodeFileId": 0,
    "seasonNumber": 1,
    "episodeNumber": 10,
    "title": "When Will Met Sylvia",
    "airDate": "2023-07-12",
    "airDateUtc": "2023-07-12T04:00:00Z",
    "runtime": 30,
    "overview": "Tensions flare between Sylvia and Will at a housewarming party.",
    "hasFile": false,
    "monitored": true,
    "unverifiedSceneNumbering": false,
    "grabbed": false,
    "id": 3984
  }
```

**Comment by causefx** [comment]
are you on orgainzr dev or master?

**Comment by causefx** [comment]
> http://localhost:8989/api/v3/calendar

can you add `&includeSeries=true` to the end of the url?

**Comment by Bencici** [comment]
> are you on orgainzr dev or master?

I was on master and I switched to dev, same issue.

> can you add `&includeSeries=true` to the end of the url?

```
[
  {
    "seriesId": 57,
    "tvdbId": 9686416,
    "episodeFileId": 0,
    "seasonNumber": 1,
    "episodeNumber": 10,
    "title": "When Will Met Sylvia",
    "airDate": "2023-07-12",
    "airDateUtc": "2023-07-12T04:00:00Z",
    "runtime": 30,
    "overview": "Tensions flare between Sylvia and Will at a housewarming party.",
    "hasFile": false,
    "monitored": true,
    "unverifiedSceneNumbering": false,
    "series": {
      "title": "Platonic (2023)",
      "sortTitle": "platonic 2023",
      "status": "continuing",
      "ended": false,
      "overview": "A platonic pair of former best friends approaching midlife reconnect after a long rift. The duo’s friendship becomes more consuming—and destabilizes their lives in a hilarious way.",
      "network": "Apple TV+",
      "airTime": "00:00",
      "images": [
        {
          "coverType": "banner",
          "remoteUrl": "https://artworks.thetvdb.com/banners/v4/series/391020/banners/641e387680f1f.jpg"
        },
        {
          "coverType": "poster",
          "remoteUrl": "https://artworks.thetvdb.com/banners/v4/series/391020/posters/644a5ab2f1e46.jpg"
        },
        {
          "coverType": "fanart",
          "remoteUrl": "https://artworks.thetvdb.com/banners/v4/series/391020/backgrounds/642dbb7ebedf0.jpg"
        }
      ],
      "originalLanguage": {
        "id": 1,
        "name": "English"
      },
      "seasons": [
        {
          "seasonNumber": 1,
          "monitored": true
        }
      ],
      "year": 2023,
      "path": "D:\\Vidéos\\SERIES\\Platonic",
      "qualityProfileId": 8,
      "seasonFolder": true,
      "monitored": true,
      "useSceneNumbering": false,
      "runtime": 30,
      "tvdbId": 391020,
      "tvRageId": 0,
      "tvMazeId": 51550,
      "firstAired": "2023-05-24T00:00:00Z",
      "seriesType": "standard",
      "cleanTitle": "platonic2023",
      "imdbId": "tt13366604",
      "titleSlug": "platonic-2023",
      "certification": "TV-MA",
      "genres": [
        "Comedy"
      ],
      "tags": [],
      "added": "2023-05-23T20:12:47Z",
      "ratings": {
        "votes": 0,
        "value": 0
      },
      "languageProfileId": 1,
      "id": 57
    },
    "grabbed": false,
    "id": 3984
  }
]
```

**Comment by causefx** [comment]
installed via docker?  if so - i can add a quick check for you to test - if not i can have you edit a file to test

**Comment by Bencici** [comment]
> installed via docker? if so - i can add a quick check for you to test - if not i can have you edit a file to test

Nope, native install.

**Comment by Healzangels** [comment]
Greetings!
I believe I am seeing the same thing on my setup.
Calendar elements for Radarr load fine but Sonarr no images.
I'm running Sonarr within Docker v4 4.0.0.645
Organizr within Docker 2.1.2430

**Comment by petercockroach** [comment]
Confirmed that adding `&includeSeries=true` worked for Sonarr for me. Do I need a different key for Radarr? I'm using Docker FWIW.

Edit: Actually that same key works for Radarr too.

**Comment by keliansb** [comment]
Just hitting the same issue for a few days now
I updated both Organizr (`latest` on `v2-master`) and Sonarr (`develop`) Docker images so I don't know which update is causing this

**Comment by keliansb** [comment]
Update : this issue has been fixed for me after upgrading Organizr to version 2.1.2460

**Comment by Bencici** [comment]
Resolved by 2.1.2460, thanks @causefx !
